packages/functions/src/resample.ts

```diff
@@ -36,24 +57,54 @@ export function resample(_options: ResampleOptions = RESAMPLE_DEFAULTS): Transfo
 		const srcAccessorCount = document.getRoot().listAccessors().length;
 		const logger = document.getLogger();
 
-		let didSkipMorphTargets = false;
+		const ready = options.ready;
+		const resample = options.resample;
+
+		await ready;
 
 		for (const animation of document.getRoot().listAnimations()) {
-			// Skip morph targets, see https://github.com/donmccurdy/glTF-Transform/issues/290.
 			const samplerTargetPaths = new Map<AnimationSampler, GLTF.AnimationChannelTargetPath>();
 			for (const channel of animation.listChannels()) {
 				samplerTargetPaths.set(channel.getSampler()!, channel.getTargetPath()!);
 			}
 
 			for (const sampler of animation.listSamplers()) {
-				if (samplerTargetPaths.get(sampler) === 'weights') {
-					didSkipMorphTargets = true;
-					continue;
-				}
-				if (sampler.getInterpolation() === 'STEP' || sampler.getInterpolation() === 'LINEAR') {
-					accessorsVisited.add(sampler.getInput()!);
-					accessorsVisited.add(sampler.getOutput()!);
-					optimize(sampler, samplerTargetPaths.get(sampler)!, options);
+				const samplerInterpolation = sampler.getInterpolation();
+
+				if (samplerInterpolation === 'STEP' || samplerInterpolation === 'LINEAR') {
+					const input = sampler.getInput()!;
+					const output = sampler.getOutput()!;
+
+					accessorsVisited.add(input);
+					accessorsVisited.add(output);
+
+					const times = toFloat32Array(input.getArray()!, input.getComponentType(), input.getNormalized());
+					const values = toFloat32Array(output.getArray()!, input.getComponentType(), input.getNormalized());
```

Comment:
If I was understanding corrently, this denormalized then input and output accessors, so should the cloned accessors be marked as `setNormalized(false)`, or should the denormalized times and values be normalized back?

Reply:
Ah good call – let's normalize the values before writing back, yes.